python/tk_desktop2/websockets/websockets_connection.py

```diff
@@ -39,6 +40,10 @@ class WebsocketsConnection(object):
     _legacy_site_warning_displayed = False
     _legacy_user_warning_displayed = False
 
+    # localhost url regex matcher
+    # https://regex101.com/r/OeTZRx/4
+    localhost_re = re.compile(r"^(?:[A-Za-z]+://)?(?:(?:shotgunlocalhost\.com)|(?:localhost))(?::\d+)?(?:/\S*)*$")
```

Comment:
I'm a bit worried about the complexity of this regular expression. Does it need to be this complicated? Also, [REDOS](https://www.owasp.org/index.php/Regular_expression_Denial_of_Service_-_ReDoS).

I would expect the origin coming from a websocket API written by us or set by the browser to follow a pretty predictable pattern, no?

Right now it seems your regular expression accepts quite a few things that we probably don't want to even support in the real world like an origin without a protocol or a port number, which will never happen.

Reply:
Naively, I would say we need to support something like [this](https://regex101.com/r/YRas6F/1)

Reply:
I can live with your approach. I will adjust two things, 
1) For some reason, the origin is http on my python client so I think we need to support http as an origin.
2) The user can decide to change the port to something bigger than 9999

Reply:
Is this better? 
https://regex101.com/r/YRas6F/2

There's no more cases where we can compute for ever.

Reply:
I'm curious, why the need for the `(?:` [This](https://regex101.com/r/7n3JIp/1) would work as well. I'll be honest I've never had to use the `(?:` before, so I'm probably missing something here.

Reply:
`( pattern )` is a capturing group
`( ?: pattern ) is a non-capturing group

Non-capturing groups are faster to evaluate.

Reply:
What about this? https://regex101.com/r/7n3JIp/2 It catches http and https, localhost and shotgunlocalhost.com, port from 1 to 5 digits.

Reply:
Ah sorry, I thought (?: was a bit more nefarious, I must have mixed it up with some other regex feature.

Yeah, let's keep your last proposition. Sorry for being pickly on something so simple.